src/query/sqs/sqs-schema-generator.service.spec.ts

```diff
@@ -0,0 +1,237 @@
+        attributes: [
+          {
+            name: 'name',
+            type: 'TEXT',
+          },
+          {
+            name: 'age',
+            type: 'INTEGER',
+          },
+        ],
+      },
+      {
+        label: 'School',
+        attributes: [
+          {
+            name: 'name',
+            type: 'TEXT',
+          },
+          {
+            name: 'type',
+            type: 'TEXT',
+          },
+          {
+            name: 'numberOfStudents',
+            type: 'INTEGER',
+          },
+        ],
+      },
+    ],
+  };
+
+  const sqsConfig = {
+    _id: '_design/sqlite:config',
+    _rev: '1-00000000',
+    sql: {
+      tables: {
+        Child: {
+          fields: {
+            name: 'TEXT',
+            age: 'INTEGER',
+            _id: 'TEXT',
+            _rev: 'TEXT',
+            created: 'TEXT',
+            updated: 'TEXT',
+            inactive: 'INTEGER',
+            anonymized: 'INTEGER',
+          },
+        },
+        School: {
+          fields: {
+            name: 'TEXT',
+            type: 'TEXT',
+            numberOfStudents: 'INTEGER',
+            _id: 'TEXT',
+            _rev: 'TEXT',
+            created: 'TEXT',
+            updated: 'TEXT',
+            inactive: 'INTEGER',
+            anonymized: 'INTEGER',
+          },
+        },
+      },
+      indexes: [],
+      options: {
+        table_name: {
+          operation: 'prefix',
+          field: '_id',
+          separator: ':',
+        },
+      },
+    },
+    language: 'sqlite',
+    configVersion:
+      '2a26f7bc7e7e69940d811a4845a5f88374cbbb9868c8f4ce13303c3be71f2ad8',
+  };
+
+  beforeEach(async () => {
+    mockCouchDbClient = {
+      changes: jest.fn(),
+      headDatabaseDocument: jest.fn(),
+      getDatabaseDocument: jest.fn(),
+      find: jest.fn(),
+      putDatabaseDocument: jest.fn(),
+    };
+
+    mockEntityConfigResolver = {
+      getEntityConfig: jest.fn(),
+    };
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        {
+          provide: SqsSchemaService,
+          useFactory: () =>
+            new SqsSchemaService(
+              mockCouchDbClient,
+              mockEntityConfigResolver,
+              sqsSchemaGeneratorConfig,
+            ),
+        },
+      ],
+    }).compile();
+
+    service = module.get<SqsSchemaService>(SqsSchemaService);
+  });
+
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+
+  it('getSchemaPath() should return Schema', () => {
+    expect(service.getSchemaPath()).toEqual('/_design/sqlite:config');
+  });
+
+  it('updateSchema() should update Schema when new sqs config version is different then current', (done) => {
+    // given
+    resolveEntityConfig();
+    resolveSqsConfigWithOtherVersion();
+    resolveDocSuccess();
+    // when
+    service.updateSchema().subscribe({
+      next: () => {
+        // then
+        expect(mockCouchDbClient.putDatabaseDocument).toHaveBeenCalled();
```

Comment:
Let's add a test (or expectation) for the structure of the actual schema being generated, also. Or maybe easiest to test and understand: test the mapToSqsSchema function?

Reply:
I added a `calledWith` check for the new schema.